test/pages/register/beneficiaries/other/mld5/UKResidentYesNoPageSpec.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.register.beneficiaries.other.mld5
+
+import models.UserAnswers
+import models.core.pages.{InternationalAddress, UKAddress}
+import org.scalacheck.Arbitrary.arbitrary
+import pages.behaviours.PageBehaviours
+import pages.register.beneficiaries.other.{AddressInternationalPage, AddressUKPage}
+
+class UKResidentYesNoPageSpec extends PageBehaviours {
+
+  "UKResidentYesNoPage" must {
+
+    beRetrievable[Boolean](UKResidentYesNoPage(0))
+
+    beSettable[Boolean](UKResidentYesNoPage(0))
+
+    beRemovable[Boolean](UKResidentYesNoPage(0))
+  }
+
+  "remove international address page when BeneficiariesAddressYesNoPage is set to true" in {
+    val index = 0
+    forAll(arbitrary[UserAnswers], arbitrary[String]) {
+      (initial, str) =>
+        val answers: UserAnswers =
+          initial.set(AddressUKPage(index), UKAddress(str, str, Some(str), Some(str), str)).success.value
+            .set(AddressInternationalPage(index), InternationalAddress(str, str, Some(str), str)).success.value
+
+        val result = answers.set(UKResidentYesNoPage(index), true).success.value
+
+        result.get(CountryOfResidencePage(index)) mustNot be(defined)
+        result.get(AddressUKPage(index)) must be(defined)
+    }
+  }
```

Comment:
This test isn't doing anything. You're asserting that CountryOfResidencePage is undefined but not setting it in the first place. This test should initially set CountryOfResidencePage, then set UKResidentYesNoPage to true, then check that CountryOfResidencePage is undefined